### DISCUSSION TITLE
SessionProvider typeError with next 15/react 19

### DISCUSSION THREAD

**Discussion by aquibbaig** [nextauthjs/next-auth#12328] [Help]
I just upgraded to Next 15/React 19 and found the following error with `next-auth@5.0.0-beta.25` & `@auth/core@0.37.4`

```
'SessionProvider' cannot be used as a JSX component.
  Its type '(props: SessionProviderProps) => Element' is not a valid JSX element type.
    Type '(props: SessionProviderProps) => Element' is not assignable to type '(props: any) => ReactNode'.
      Type 'Element' is not assignable to type 'ReactNode'.
```

### Usage
```
'use client'
...
<SessionProvider>
   <ThemeProvider attribute="class" enableSystem disableTransitionOnChange>
      ...
    </ThemeProvider>
</SessionProvider>
```

**Comment by pepew-le-boss** [comment]
same error

**Comment by gadcam** [comment]
Actually it looks like it is a known issue we can see this in the code
https://github.com/nextauthjs/next-auth/blob/507aaddaa7cd6f770038b0c4a55579bfc141e999/packages/next-auth/src/react.tsx#L533-L535


I think we have two workaround for this
- ignore the error with
```ts
  // @ts-expect-error - Known issue with Auth.js v5 + React 19 type definitions
```
- add a `global.d.ts` file at the root of the repository
```ts
import { Element } from 'react';

declare global {
  namespace JSX {
    // TOFIX: Remove this hack
    // Hack to allow `React.JSX.Element` to be used as a type
    interface Element extends React.JSX.Element {}
  }
}
```
I would rather go with the first option as it is more targeted.
But if you have this issue all over your codebase it might be a good thing to start with the `global.d.ts` solution to unlock the situation.
This might be linked with https://react.dev/blog/2024/04/25/react-19-upgrade-guide#the-jsx-namespace-in-typescript
